lib/elastic_apm/config.rb

```diff
@@ -281,11 +298,21 @@ def set_from_env
 
     def set_from_args(options)
       assign(options)
+    rescue ConfigError => e
+      alert_logger.warn format(
+        'Failed to configure from arguments: %s',
+        e.message
+      )
     end
 
     def set_from_config_file
       return unless File.exist?(config_file)
       assign(YAML.load_file(config_file) || {})
+    rescue ConfigError => e
+      alert_logger.warn format(
+        'Failed to configure from config file: %s',
+        e.message
+      )
```

Comment:
In the Sinatra app this leads to e.g.
```
W, [2018-12-04T12:59:51.383087 #97702]  WARN -- The option `debug_transactions=' has been removed.: [ElasticAPM]
W, [2018-12-04T12:59:51.383260 #97702]  WARN -- The option `enabled_environments=' has been removed.: [ElasticAPM]
W, [2018-12-04T12:59:51.383342 #97702]  WARN -- Failed to configure from arguments: No such option `nonexist': [ElasticAPM]
```

There is no `error` logged but APM doesn't start. I suggest to log this as an `error` instead of a `warning`.

Reply:
Right it did start, I checked for documents in ES, but there were non so I thought it didnt. But it actually seems I caught an unrelated issue.